crates/subspace-networking/src/node.rs

```diff
@@ -200,9 +234,15 @@ impl Node {
             shared,
             is_relay_server,
             relay_server_memory_port: Arc::new(Mutex::new(None)),
+            sync_status: Arc::new(NodeSynchronizationStatusHandler::new()),
         }
     }
 
+    /// Node's synchronization status handler.
+    pub fn sync_status_handler(&self) -> NodeSynchronizationStatusHandler {
+        (*self.sync_status).clone()
```

Comment:
Why do you wrap in an `Arc` then?

Reply:
This was resolved, but still looks the same as before. Can you explain?

Reply:
Not really, I remember fixing it but I can't find the changes even in my local history. It seems I discarded them before the commit. Sorry for a confusion, I'm adding an additional commit for this fix.